Project-Hotpot/Project-Hotpot/Managers/SpotifyManager.swift

```diff
@@ -145,29 +148,35 @@ extension SpotifyManager: SPTSessionManagerDelegate {
 
 // MARK: - Networking
 extension SpotifyManager {
-    func makeSpotifyRequest (request: URLRequest, completion: @escaping ([String: Any]?, Error?) -> Void) {
+    func makeSpotifyRequest (request: URLRequest, completion: @escaping (_ result: Result<[String: Any], Error>) -> Void) {
         let task = URLSession.shared.dataTask(with: request) { data, response, error in
             guard let data = data,                              // is there data
                   let response = response as? HTTPURLResponse,  // is there HTTP response
                   (200 ..< 300) ~= response.statusCode,         // is statusCode 2XX
-                  error == nil else {                           // was there no error, otherwise ...
-                NSLog("Error fetching token \(error?.localizedDescription ?? "")")
-                return completion(nil, error)
+                  error == nil
+            else {// was there no error, otherwise ...
+                if let error = error {
+                    NSLog("Error fetching token \(error.localizedDescription)")
+                    return completion(.failure(error))
+                }
```

Comment:
This seems like a bit of a hefty `guard` statement. Consider first checking if you have an error and handling that separately, *then* jumping into data, response, and status codes (and handling those subsequent errors as needed).